src/internal/connector/exchange/api/contacts.go

```diff
@@ -190,12 +190,92 @@ func (c Contacts) EnumerateContainers(
 var _ itemPager = &contactPager{}
 
 type contactPager struct {
+	gs      graph.Servicer
+	builder *users.ItemContactFoldersItemContactsRequestBuilder
+	options *users.ItemContactFoldersItemContactsRequestBuilderGetRequestConfiguration
+}
+
+func NewContactPager(
+	ctx context.Context,
+	gs graph.Servicer,
+	user, directoryID string,
+	immutableIDs bool,
+) (itemPager, error) {
+	options, err := optionsForContactFoldersItem([]string{"parentFolderId"}, immutableIDs)
+	if err != nil {
+		return &contactPager{}, err
+	}
+
+	builder := gs.Client().UsersById(user).ContactFoldersById(directoryID).Contacts()
+
+	if len(os.Getenv("CORSO_URL_LOGGING")) > 0 {
+		gri, err := builder.ToGetRequestInformation(ctx, options)
+		if err != nil {
```

Comment:
should we be returning this error somehow? How will callers know that they can't use the pager? Is the pager going to check later if it was setup properly?

this would apply to other places this pattern is used as well

Reply:
The current version just logs them as well. I believe we should be fine as this data is only used for logging and if the endpoint actually fails, we will fail.
https://github.com/alcionai/corso/blob/edef23bfc4eb57acc7ced0a36830a58d76acbf15/src/internal/connector/exchange/api/mail.go#L364